osbs/build/user_params.py

```diff
@@ -0,0 +1,207 @@
+
+        self.platforms.value = platforms
+        self.platform.value = platform
+        self.koji_target.value = koji_target
+        self.koji_task_id.value = koji_task_id
+        self.filesystem_koji_task_id.value = filesystem_koji_task_id
+        self.koji_parent_build.value = koji_parent_build
+        self.flatpak.value = flatpak
+        self.flatpak_base_image.value = flatpak_base_image
+        self.isolated.value = isolated
+        self.scratch.value = scratch
+
+        if not flatpak:
+            if not base_image:
+                raise OsbsValidationException("base_image must be provided")
+            self.trigger_imagestreamtag.value = get_imagestreamtag_from_image(base_image)
+
+            if not name_label:
+                raise OsbsValidationException("name_label must be provided")
+            self.imagestream_name.value = name_label.replace('/', '-')
+
+        if signing_intent and compose_ids:
+            raise OsbsValidationException(
+                'Please only define signing_intent -OR- compose_ids, not both')
+        if compose_ids and yum_repourls:
+            raise OsbsValidationException(
+                'Please only define yum_repourls -OR- compose_ids, not both')
+        if not (compose_ids is None or isinstance(compose_ids, list)):
+            raise OsbsValidationException("compose_ids must be a list")
+        if not (yum_repourls is None or isinstance(yum_repourls, list)):
+            raise OsbsValidationException("yum_repourls must be a list")
+        self.yum_repourls.value = yum_repourls or []
+        self.signing_intent.value = signing_intent
+        self.compose_ids.value = compose_ids or []
+
+        if additional_tag_data:
+            self.additional_tags = AdditionalTagsConfig(additional_tag_data['dir_path'],
+                                                        additional_tag_data['file_name'],
+                                                        additional_tag_data['tags'])
+        self._populate_image_tag()
+
+    def __repr__(self):
+        return "UserParams(%s)" % self.__dict__
+
+    def from_json(self, user_params_json):
+        if not user_params_json:
+            return
+        json_dict = json.loads(user_params_json)
+        logger.debug("CALLED FROM JSON HERE with dict %s", json_dict)
+        for key, value in json_dict.items():
+            try:
+                self.convert_dict[key].value = value
+            except KeyError:
+                continue
+        if 'additional_tags' in json_dict.keys():
+            logger.debug("read additional tags %s from json", json_dict['additional_tags'])
+            self.additional_tags = AdditionalTagsConfig(tags=json_dict['additional_tags'])
+
+    def set_if_exists(self, json_dict, param):
+        if self.convert_dict[param].value:
+            json_dict[param] = self.convert_dict[param].value
+
+    def to_json(self):
+        json_dict = {}
+        for key in self.convert_dict:
```

Comment:
What if we just iterate through object attributes and only operate on `BuildParam` types?
```python
for x, param in b.__dict__.items():
    if isinstance(param, BuildParam):
        print x, param.value
```

Reply:
Fixed I think.